hw/ip/otbn/dv/verilator/otbn_top_sim.cc

```diff
@@ -0,0 +1,57 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <fstream>
+#include <iostream>
+#include <iomanip>
+#include <svdpi.h>
+
+#include "verilated_toplevel.h"
+#include "verilator_memutil.h"
+#include "verilator_sim_ctrl.h"
+
+extern "C" {
+extern unsigned int otbn_base_reg_get(int index);
+}
+
+int main(int argc, char **argv) {
+  otbn_top_sim top;
+  VerilatorMemUtil memutil;
+  VerilatorSimCtrl &simctrl = VerilatorSimCtrl::GetInstance();
+  simctrl.SetTop(&top, &top.IO_CLK, &top.IO_RST_N,
+                 VerilatorSimCtrlFlags::ResetPolarityNegative);
+
+  memutil.RegisterMemoryArea(
+      "dmem", "TOP.otbn_top_sim.u_dmem.u_mem.gen_generic.u_impl_generic");
+  memutil.RegisterMemoryArea(
+      "imem", "TOP.otbn_top_sim.u_imem.u_mem.gen_generic.u_impl_generic");
+  simctrl.RegisterExtension(&memutil);
+
+  bool exit_app = false;
+  int ret_code = simctrl.ParseCommandArgs(argc, argv, exit_app);
+  if (exit_app) {
+    return ret_code;
+  }
+
+  std::cout << "Simulation of OTBN" << std::endl
+            << "==================" << std::endl
+            << std::endl;
+
+  simctrl.RunSimulation();
+
+  if (!simctrl.WasSimulationSuccessful()) {
+    return 1;
+  }
+
+  svSetScope(svGetScopeFromName("TOP.otbn_top_sim"));
+  std::cout << "Final Register Values:" << std::endl;
```

Comment:
This final register dump can probably go fairly soon however I think it's useful for quick tests as it avoids the need for opening GTKWave to check what happened.

Reply:
Let's leave it in until we have tracing.